rpc/src/v1/impls/parity.rs

```diff
@@ -417,6 +510,29 @@ impl<C, M, U, S> Parity for ParityClient<C, M, U> where
 		}))
 	}
 
+		fn raw_block_by_number(&self, num: BlockNumber, include_txs: bool) -> BoxFuture<Option<RichBlock>> {
+			Box::new(future::done(self.rich_block(num.into(), include_txs)))
+		}
+
+		fn submit_block(&self, block: RichBlock) -> Result<RpcH256> {
+			// TODO: Re-implement rlp_bytes for rpc/v1/src/types/block.rb 
+			//       or reuse the rlp_bytes from ethcore/src/block.rb?
+			rlp_encoded_block = block.rlp_bytes()
+			Rlp::new(&rlp_encoded_block.into_vec()).as_val()
+				.map_err(errors::rlp)
+				.and_then(|tx| SignedTransaction::new(tx).map_err(errors::transaction))
+				.and_then(|signed_transaction| {
+					FullDispatcher::dispatch_transaction(
```

Comment:
This wraps the block RLP and dispatches it as if it were a transaction, whereas in this case it is a full block. We should validate the block against our chain and then import it locally using the miner (which should trigger propagation afterwards).

I think we should validate the block and import it locally (this requires that the client is synced for this method to work) instead of blindingly broadcasting it, since otherwise this endpoint can be used to easily spam the network with invalid blocks.

Reply:
@ppratscher is there something in the current codebase that already does this or can be adapted to accept the block RLP? The code block within `impl Rebuilder for PowRebuilder` (starting at line 220 in ethcore/src/snapshot/consensus/work.rs) seems to have some relevance to this, but I'm unsure if I'm missing the mark on this one. Appreciate any further guidance you can give on this one!